src/components/stack-chart/index.js

```diff
@@ -90,30 +99,41 @@ class StackChartGraph extends React.PureComponent<Props> {
   }
 
   _onSelectedCallNodeChange = (
-    callNodeIndex: IndexIntoCallNodeTable | null
+    callNodeorMarkerIndex: IndexIntoCallNodeTable | null
   ) => {
     const { callNodeInfo, threadIndex, changeSelectedCallNode } = this.props;
     changeSelectedCallNode(
       threadIndex,
-      getCallNodePathFromIndex(callNodeIndex, callNodeInfo.callNodeTable)
+      getCallNodePathFromIndex(
+        callNodeorMarkerIndex,
+        callNodeInfo.callNodeTable
+      )
     );
   };
 
   _onRightClickedCallNodeChange = (
-    callNodeIndex: IndexIntoCallNodeTable | null
+    node: { type: 'call-node' | 'marker', index: IndexIntoCallNodeTable } | null
   ) => {
     const {
       callNodeInfo,
       threadIndex,
       changeRightClickedCallNode,
+      changeRightClickedMarker,
     } = this.props;
-    changeRightClickedCallNode(
-      threadIndex,
-      getCallNodePathFromIndex(callNodeIndex, callNodeInfo.callNodeTable)
-    );
+
+    if (node.type === 'marker') {
+      changeRightClickedMarker(threadIndex, node.index);
+    } else {
+      changeRightClickedCallNode(
+        threadIndex,
+        getCallNodePathFromIndex(node.index, callNodeInfo.callNodeTable)
+      );
```

Comment:
What do you think of this initial strategy for setting the appropriate right clicked node?

It looks like it should work, but for some reason i'm not getting the marker context menu on the stack chart. Do you have any guesses of what could be going wrong?

Reply:
I like the way an object is used to signal the type of the thing clicked, I think that will generalize well, as long as it's not triggering spurious updates.

> It looks like it should work, but for some reason i'm not getting the marker context menu on the stack chart. Do you have any guesses of what could be going wrong?

Oof. This is because, the trigger is only set up for the `CallNodeContextMenu`. I'd say we can probably file a follow-up, but this would require a bit more re-architecting to get to work. Essentially we'd need to consolidate under one context menu for both context menu types. I don't think it's worth doing that work here, since it's a fairly big amount of work. I'm fine with just filing a follow-up for it.

```
          <ContextMenuTrigger
            id="CallNodeContextMenu"
```